### PR TITLE
Adds initial roles

### PR DESCRIPTION
**Pull request by JPrevost**
## Status
**READY**

#### What does this PR do?

Adds initial roles and tests to confirm expectations.

#### Helpful background context (if appropriate)

https://github.com/CanCanCommunity/cancancan

#### How can a reviewer manually see the effects of these changes?

The PR builds are broken (ticket is open) but if you look at this locally you can create a couple users and give one a thesis and confirm the other can't see it.

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/TI-6

#### Todo:
- [x] Tests
- [x] Documentation
- [ ] Stakeholder approval

#### Requires Database Migrations?
NO

#### Includes new or updated dependencies?
YES

### CODE REVIEW

File: test/controllers/thesis_controller_test.rb [hunk 1]
```diff
@@ -58,4 +58,78 @@ class ThesisControllerTest < ActionDispatch::IntegrationTest
...
+    assert_response :redirect
+  end
+
+  test 'admin users can view another user thesis' do
+    sign_in users(:admin)
+    get "/thesis/#{theses(:one).id}"
+    assert_response :success
+  end
+
+  # Tests below this note are to protect us from our future selves.
```
Comment (thatandromeda) [inline test/controllers/thesis_controller_test.rb]: 👍

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  def initialize(user)
+    # Define abilities for the passed in user here.
+    # See the documentation for details:
+    # https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities
+
+    # Any user can create a new Thesis
+    can :create, Thesis
+
+    # Only the Thesis owner can view their Thesis
+    can :read, Thesis, user_id: user.id
```
Comment (thatandromeda) [inline app/models/ability.rb]: The docs wrap tests of user attributes in an `if user.present?`, but this seems to work even if we don't do that. I guess it's cool? (ISTR rails and django handle anonymous users differently but I don't follow the specifics.)

Comment (JPrevost) [reply app/models/ability.rb]: oh. right. Let me dig in on that a bit more as now I'm not sure...hmmm.

**Comment by JPrevost** [comment]
I've updated abilities.rb to match the documentation. Thanks for the catch @thatandromeda

**Pull request merged**